### PR TITLE
[BEAM-354] Move V1Beta3 DatastoreSource into its own file

### PR DESCRIPTION
**Pull request by vikkyrk**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

- No logic changes, just move DataSource to new file and rename to V1Beta3Source

### CODE REVIEW

**Comment by vikkyrk** [comment]
@dhalperi A small PR, breaking my earlier change into parts :)

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/V1Beta3Source.java [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import javax.annotation.Nullable;
+
+/**
+ * A Datstore Source for v1beta3 API.
+ */
+public class V1Beta3Source extends BoundedSource<Entity>{
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/V1Beta3Source.java]: not `public`

**Comment by dhalperi** [comment]
Eugene had a good suggestion: why not

Create.of(query) | ParDo.of(split query) | Reshard | ParDo.of(read from query)

Your code will probably be much simpler and we can let the dynamic work rebalancing happen on the GroupByKey inside of the reshard.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/V1Beta3Source.java (hunk 1 above)
Comment (vikkyrk) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/V1Beta3Source.java]: Fixed

**Comment by vikkyrk** [comment]
Interesting. Let me move that suggestion to the main RB where I implemented dyn rebalance. This change will be relevant either way since I am going to cleanup the mock stuff.

**Comment by vikkyrk** [comment]
No longer relevant

**Pull request closed without merging**